src/languages/cpp.js

```diff
@@ -423,15 +431,11 @@ export default function(hljs) {
       _hint: FUNCTION_HINTS },
     begin: regex.concat(
       /\b/,
-      /(?!decltype)/,
-      /(?!if)/,
-      /(?!for)/,
-      /(?!switch)/,
-      /(?!while)/,
+      // adds all keywords dynamically
+      ...allKeywords.map(keyword => new RegExp(`(?!${keyword})`)),
```

Comment:
Actually don't we only care about keywords here that can potentially have a `(` following them?  That's not ALL keywords, is it?

Reply:
We can ignore this for now since I already said this was ready but need your author link...